README.md

```diff
@@ -1,173 +1,243 @@
 
-As of 10/25/2024, unfortunately Google Colab does not yet support Python 3.11 or 3.12
-([issue](https://github.com/googlecolab/colabtools/issues/3190)).
+```bash
+pip install 'fhaviary[gsm8k]'
+pip install 'fhaviary[hotpotqa]'
+pip install 'fhaviary[litqa]'
+pip install 'fhaviary[lfrqa]'
+```
 
-Thus, as a workaround, you will need to install Python 3.11 into your notebook.
-Here is a sample notebook showing how to do this:
-https://colab.research.google.com/drive/1mejZ5cxgKZrMpYEe0iRoanaGGQ0Cr6WI?usp=sharing
+To run the tutorial notebooks:
 
-Also, note that `async` code works in Google Colab.
+```bash
+pip install "fhaviary[dev]"
+```
 
 ### Developer Installation
 
-For local development, please see the [CONTRIBUTING.md](CONTRIBUTING.md).
+For local development, please see [CONTRIBUTING.md](CONTRIBUTING.md).
 
-## Messages
+## Tutorial Notebooks
 
-Communication between the agent and environment is done through messages.
-Messages have two attributes:
+Check out our tutorial notebooks.
 
-```py
-msg = Message(content="Hello, world!", role="assistant")
-```
+## Defining a Custom Environment
 
-The `content` is a string with a text, a JSON serializable list of `dict`s, or a null value.
-A list of dicts is used to encode multi-modal content. The method `create_message` can be used to create a message with images:
+The example below walks through defining a custom language agent environment in Aviary.
+We define a simple environment where an agent takes actions to modify a counter.
+The example is also featured in the following notebook TODO: Fill in.
 
-```py
-from PIL import Image
-import numpy as np
+```python
+from collections import namedtuple
+from aviary.core import Environment, Message, ToolRequestMessage, Tool
 
-img = Image.open("your_image.jpg")
-img_array = np.array(img)
+# State in this example is simply a counter
+CounterEnvState = namedtuple("CounterEnvState", ["count"])
 
-msg = Message.create_message(role="user", text="Hello, world!", images=[img_array])
-```
 
-`create_message` supports images as numpy array or base64 encoded images. In this case, `content` will be a list of dictionaries with the keys `text` and `image_url`.
+class CounterEnv(Environment[CounterEnvState]):
+    """A simple environment that allows an agent to modify a counter."""
 
-```py
-{
-    {"type": "text", "text": "Hello World!"},
-    {"text": "image_url", "image_url": "data:image/png;base64,{base64_image}"},
-}
+    async def reset(self):
+        """Initialize the environment with a counter set to 0."""
+        self.state = CounterEnvState(count=0)
+
+        # Create tools allowing the agent to increment and decrement counter
+        self.tools = [
+            Tool.from_function(self.incr),
+            Tool.from_function(self.decr),
+        ]
+
+        # Return an observation message with the counter and available tools
+        return [Message(content=f"counter={self.state.count}")], self.tools
```

Comment:
Normally `reset` comes with some quest -- there is no quest here tho

It should be like "Count to 10" or something

Reply:
```suggestion
        return [Message(content=f"Count to 10. counter={self.state.count}")], self.tools
```

Reply:
I think you updated in the `docs` Ipynb but not in the `README` port